### PR TITLE
[LW-10004] add deleagtion tx

### PR DESCRIPTION
**Pull request by VanessaPC**
# Context
- We need a way to single delegate / undelegate to be able to see deposits in dapp connector

# Proposed Solution
- add delegation and undelegation

# Important Changes Introduced

### CODE REVIEW

File: packages/dapp-connector-client/package.json
Comment (mkazlauskas) [inline packages/dapp-connector-client/package.json:55]: What is the fix here? :thinking:

File: packages/e2e/test/web-extension/dapp-sdk/features/singleDelegation.ts
Comment (mkazlauskas) [inline packages/e2e/test/web-extension/dapp-sdk/features/singleDelegation.ts:31]: It shouldn't matter whether wallet has any native assets for delegation. Please remove the assertion or update to check coins. Same for undelegation function.
```suggestion
        if (availableBalance.coins === 0n) {
          throw new Error('Your wallet has no coins');
        }
```

File: packages/e2e/test/web-extension/dapp-sdk/features/singleDelegation.ts
Comment (mkazlauskas) [inline packages/e2e/test/web-extension/dapp-sdk/features/singleDelegation.ts:27]: We can trust the types since this is not nullable, and we're making this assertion in dapp-sdk.ts already. Same for undelegation and sendSeveralAssets.
```suggestion
```

File: packages/e2e/test/web-extension/dapp-sdk/features/singleUndelegation.ts
Comment (mkazlauskas) [inline packages/e2e/test/web-extension/dapp-sdk/features/singleUndelegation.ts:14]: This `combineLatest` is redundant, since we're using only 1 observable
```suggestion
  connectedWallet.balance.utxo.available$
```

File: packages/e2e/test/web-extension/dapp-sdk/features/singleUndelegation.ts
Comment (mkazlauskas) [inline packages/e2e/test/web-extension/dapp-sdk/features/singleUndelegation.ts:28]: This is a little bit confusing, as we're passing an already 'builtTx' to 'buildAndSignTx'.
```suggestion
        const txBuilder = connectedWallet.createTxBuilder().delegatePortfolio(null);

        buildAndSignTx({ txBuilder, connectedWallet, textElement: undelegateAssetsElement });
```

**Review by mkazlauskas — CHANGES_REQUESTED** [review]
Looks good, just some minor change requests below.

Also please squash 9828c6eb756d5015570475da8808ef63764497d4 to 78aa343dd61af1d89ed34ca513980d9cef17be0e before merging

File: packages/dapp-connector-client/package.json
Comment (VanessaPC) [reply packages/dapp-connector-client/package.json:55]: None, but this came from one of the merges, I'll revert.

File: packages/e2e/test/web-extension/dapp-sdk/features/singleDelegation.ts
Comment (VanessaPC) [reply packages/e2e/test/web-extension/dapp-sdk/features/singleDelegation.ts]: it's addressed here: ac7ca1e3962ca8240a8e701b545712cc53127eaa

File: packages/e2e/test/web-extension/dapp-sdk/features/singleUndelegation.ts
Comment (VanessaPC) [reply packages/e2e/test/web-extension/dapp-sdk/features/singleUndelegation.ts]: it's addressed here: ac7ca1e3962ca8240a8e701b545712cc53127eaa

File: packages/e2e/test/web-extension/dapp-sdk/features/singleUndelegation.ts
Comment (VanessaPC) [reply packages/e2e/test/web-extension/dapp-sdk/features/singleUndelegation.ts]: it's addressed here: ac7ca1e3962ca8240a8e701b545712cc53127eaa , although, instead of renaming the txBuilder, I am renaming the function since it's not building but inspecting and signing, to `inspectAndSingTx`

**Review by mkazlauskas — APPROVED** [review]
Looks good! There are some [build errors](https://github.com/input-output-hk/cardano-js-sdk/actions/runs/8295344034/job/22702179212?pr=1145#step:4:927) though

**Comment by DominikGuzei** [comment]
@VanessaPC it works and i see a build error for the test Dapp:
image

**Comment by mkazlauskas** [comment]
@DominikGuzei did you rebuild all packages? PersonalWallet was renamed to BaseWallet. If the issue is in dapp-connector-client package, please fix in a separate commit so that we can squash it after it's merged in the base branch.

**Comment by DominikGuzei** [comment]
@mkazlauskas yes, i did `yarn & yarn build` in the root before building the test Dapp

**Comment by DominikGuzei** [comment]
> Looks good! There are some [build errors](https://github.com/input-output-hk/cardano-js-sdk/actions/runs/8295344034/job/22702179212?pr=1145#step:4:927) though

I think that might be due to the e2e package being built before its dependencies? it works fine if you build all the packages and then try again.

**Comment by DominikGuzei** [comment]
@mkazlauskas also, nothing happens when i click "delegate" in the test Dapp - i used a preprod wallet which already has one delegation portfolio to one pool. Is that an issue, should it be a clean slate in order to work?

**Comment by VanessaPC** [comment]
> @VanessaPC @mkazlauskas also, nothing happens when i click "delegate" in the test Dapp - i used a preprod wallet which already has one delegation portfolio to one pool. Is that an issue, should it be a clean slate in order to work?

I believe this is fixed @DominikGuzei , @mkazlauskas I will fix the build, I'm currently looking at another bug on another branch.

**Review by DominikGuzei — APPROVED** [review]
i don't want to block this PR, seems like the issue i ran into is on my side 👍

**Pull request merged**
